server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java

```diff
@@ -101,7 +101,8 @@ public void execute(SearchContext context) {
             if (!context.hasScriptFields() && !context.hasFetchSourceContext()) {
                 context.fetchSourceContext(new FetchSourceContext(true));
             }
-            fieldsVisitor = new FieldsVisitor(context.sourceRequested());
+            boolean loadSource = context.sourceRequested() || context.fetchFieldsContext() != null;
```

Comment:
Should we have the same logic in the `else` below or do we expect users to not use `stored_fields` in conjunction with the new `fields` option ?

Reply:
I think it'd be safest to add it below too.

Reply:
Oops, thanks for catching this. I'll push an update.